src/Avalanche/Signer.cpp

```diff
@@ -0,0 +1,353 @@
+        }
+        case Proto::TransactionOutput::kNftTransferOutput: {
+            auto nftTxferStruct = outputStruct.nft_transfer_output();
+            auto groupID = nftTxferStruct.group_id();
+            auto payload = Data(nftTxferStruct.payload().begin(), nftTxferStruct.payload().end());
+            auto locktime = nftTxferStruct.locktime();
+            auto threshold = nftTxferStruct.threshold();
+            auto addresses = structToAddresses(nftTxferStruct.addresses());
+            
+            return std::make_unique<NFTTransferOutput>(groupID, payload, locktime, threshold, addresses);
+        }
+        case Proto::TransactionOutput::kNftMintOutput: 
+        default: {
+            auto nftMintStruct = outputStruct.nft_mint_output();
+            auto groupID = nftMintStruct.group_id();
+            auto locktime = nftMintStruct.locktime();
+            auto threshold = nftMintStruct.threshold();
+            auto addresses = structToAddresses(nftMintStruct.addresses());
+
+            return std::make_unique<NFTMintOutput>(groupID, locktime, threshold, addresses);
+        }
+    } // end switch-case deciding which output struct to build
+}
+
+std::vector<TransferableOutput> structToOutputs(const google::protobuf::RepeatedPtrField<TW::Avalanche::Proto::TransferableOutput> outputStructs) noexcept {
+    std::vector<TransferableOutput> outputs;
+    for (auto &outputStruct : outputStructs) {
+        auto assetID = Data(outputStruct.asset_id().begin(), outputStruct.asset_id().end());
+        auto txnOut = extractTransferOut(outputStruct.output());
+        auto txferOut = TransferableOutput(assetID, std::move(txnOut));
+        outputs.push_back(txferOut);
+    } // end for loop building output structs
+    return outputs;
+}
+
+std::vector<std::unique_ptr<TransactionOutput>> extractOutputsFromInitialState(TW::Avalanche::Proto::InitialState stateStruct) noexcept {
+    std::vector<std::unique_ptr<TransactionOutput>> outputs;
+    for (auto &outputStruct : stateStruct.outputs()) {
+        auto txnOut = extractTransferOut(outputStruct);
+        outputs.push_back(std::move(txnOut));
+    } // end for loop building output structs
+    return outputs;
+}
+
+std::vector<TransferableOp> structToOperations(google::protobuf::RepeatedPtrField<TW::Avalanche::Proto::TransferableOp> opsStruct) noexcept {
+    std::vector<TransferableOp> ops;
+    for (auto &op : opsStruct) {
+        auto assetID = Data(op.asset_id().begin(), op.asset_id().end());
+        std::vector<TransferableOp::UTXOID> utxoIDs;
+        for (auto &utxoidStruct: op.utxo_ids()) {
+            auto txid = Data(utxoidStruct.tx_id().begin(), utxoidStruct.tx_id().end());
+            auto utxoIndex = utxoidStruct.utxo_index();
+            auto utxoid = std::make_pair(txid, utxoIndex);
+            utxoIDs.push_back(utxoid);
+        } // end for loop building utxo ids
+        switch (op.transfer_op().operation_case()) {
```

Comment:
warning: enumeration value 'OPERATION_NOT_SET' not handled in switch
no default